testapp/src/androidTest/java/com/neuroid/example/EventRecorder.kt

```diff
@@ -0,0 +1,38 @@
+package com.neuroid.example
+
+import junit.framework.TestCase.assertTrue
+
+/**
+ * record events received from the app through the mock webserver and
+ * verify them for event integrity.
+ */
+class EventRecorder {
+    private val eventBuffer = mutableListOf<EventModel>()
+
+    fun addEvent(eventModel: EventModel) {
+        eventBuffer.add(eventModel)
+    }
+
+    /**
+     * for now we will simply count the number of events and compare
+     * with an expected event count. In the future we should pass in a
+     * JSON expected result and compare with the events received from
+     * the app.
+     */
+    fun verifyEventList(expectedEventCount: Int) {
+        var eventCount = 0
+        println("printEventList --------start----------")
+        eventBuffer.forEach { eventModel ->
+            eventModel.jsonEvents.forEach { jsonEventModel ->
+                eventCount ++
+            }
+        }
+        println("printEventList--------end count: expected: $expectedEventCount, actual: $eventCount ----------")
+        // some emulators will returns different events counts, we can approximate this check to
+        // account for the variance here and get a ballpark result here
+        assertTrue("expectedEventCount: ${expectedEventCount} ~ eventCount: ${eventCount} +-20 ",
+            (eventCount >= expectedEventCount - 20 || eventCount <= expectedEventCount + 20)
```

Comment:
Is it worth putting the 20 in a variable we could more easily control?

Reply:
Yeah I can put make em constants. thanks.

Reply:
I think I need the eval to be && instead of || as well.

Reply:
Running awesome now!